c7n/resources/s3.py

```diff
@@ -1493,40 +1481,27 @@ def process(self, buckets):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        state = self.data.get('state', 'enable')
-        scopes = self.data.get('scope', ['All'])
+        config = {}
         try:
             config = s3.get_public_access_block(
                 Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
         except ClientError as e:
             if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else statement below
-                config = None
+                # When there is a None config, set all to false to be symbolic
+                config = {key: False for key in self.keys}
```

Comment:
this initialization should be at the beginning of the method, not in the error handler.